Rules/UseCmdletCorrectly.cs

```diff
@@ -129,28 +176,25 @@ private bool MandatoryParameterExists(CommandAst cmdAst)
                 return true;
             }
 
-            if (mandParams.Count == 0 || (Helper.Instance.IsKnownCmdletFunctionOrExternalScript(cmdAst) && Helper.Instance.PositionalParameterUsed(cmdAst)))
+            if (mandatoryParameters.Count == 0)
             {
-                returnValue = true;
+                return true;
             }
-            else
+
+            // Compares parameter list and mandatory parameter list.
+            foreach (CommandElementAst commandElementAst in cmdAst.CommandElements.OfType<CommandParameterAst>())

```

Comment:
Worth noting that the mandatory parameter logic here is pretty flawed (I haven't changed it from the original). It takes a list of all the mandatory parameters of the cmdlet and checks to see that at least one was supplied.

But:

- There could be a parameter set specified by a provided non-mandatory parameter which has no mandatory parameters
- There could be multiple mandatory parameters in a given parameter set

Parameter binding is hard and rather than trying to fix the logic here, any attempt to fix it should be more general-purpose. But nobody seems to have complained about this yet, so I doubt it's actually a huge issue.

Reply:
It's something I'd like to fix in an external library and reuse in PSSA, PSES and a few other places